### DISCUSSION TITLE
v7.0.0

### DISCUSSION THREAD

**Discussion by tadashi-aikawa** [tadashi-aikawa/obsidian-various-complements-plugin#133] [Announcements]
I have started to develop v7.0.0. It will contain many new features and many breaking changes in settings.

### **Chinese strategy** (#7) ( `🏷️beta1` )

**`cedict_ts.u8` is required**

https://user-images.githubusercontent.com/9500018/170834259-5579d628-8f3e-46c8-8e41-c92bde8a0b36.mp4

![image](https://user-images.githubusercontent.com/9500018/170834278-6948ae0c-203a-4103-9869-6d76413fc852.png)

- Fix a bug that can't install on mobile ( `🏷️beta2` )
- Fix a bug that suggestion board doesn't disappear when press space (Chinese strategy) (#136) ( `🏷️beta7` )

### **Respect to "New link format" option** (#116) ( `🏷️beta4` )

⚠️ **Breaking changes**

![image](https://user-images.githubusercontent.com/9500018/170856039-128fac4c-06f8-42ef-a2d6-a18d9ec09383.png)

https://user-images.githubusercontent.com/9500018/170856071-0658fc29-5bba-42e9-aeeb-15214ceff2f3.mp4

- Fix a bug that doesn't complete the internal link whose file doesn't exist (#116)
 ( `🏷️beta6` )

### **Only suggest words longer than X characters** (#125)  ( `🏷️beta5` )

⚠️ **Breaking changes**

It can set `Min number of characters for indexing` options in [Current file complement] and [Current vault complement].

https://user-images.githubusercontent.com/9500018/170860401-414d7bae-70b1-4395-aaec-5aa1e1acdbbb.mp4

Default value are (Just because)

| Strategy | default value |
|-|-|
| default | 5 |
| english-only | 5 |
| japanese | 2 |
| arabic | 3 |
| chinese | 2 |

## ✨ **New things**

- Use a case-insensitive match instead of a case-sensitive (only first letter) match for the following features (#101) ( `🏷️beta3` )
    - [Current file complement]
    - [Current vault complement]
    - [Custom dictionary complement](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/1.%20Features/Custom%20dictionary%20complement/#matching-logic)
- Add [Complement automatically] status to the [Status bar] (#122) ( `🏷️beta8` )
- Add "Shift+Space", "Backquote", and "None" to a [Select a suggestion key] option (#128) ( `🏷️beta8` )
- Add a "Disable the up/down keys for cycle through suggestions keys" option (#129) ( `🏷️beta8` )
    - Fix a bug that doesn't hide suggestions in the edge case ( `🏷️beta9` )

## 🦾 **No longer broken**

- todo

## 🔥 **Breaking changes**

- todo

## 🚑 **Hot fix**

- todo

[Current file complement]: https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/1.%20Features/Current%20file%20complement/#matching-logic
[Current vault complement]: https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/1.%20Features/Current%20vault%20complement/#matching-logic
[Status bar]: https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/5.%20Terms/%F0%9F%96%BC%EF%B8%8FStatus%20bar/
[Select a suggestion key]: https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.3.%20Key%20customization/%E2%9A%99%EF%B8%8FSelect%20a%20suggestion%20key/
[Complement automatically]: https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.1.%20Main/%E2%9A%99%EF%B8%8FComplement%20automatically/

**Comment by aubreyz** [comment]
This is VERY exciting.

**Comment by woshitoutouge** [comment]
OMG Chinese! Absolutely great work!

**Comment by tadashi-aikawa** [comment]
I will release v7.0.0 in this night 🍷  (About 2022-06-05T14:00 UTC)
